lib/commands/instrument.js

```diff
@@ -92,40 +97,45 @@ exports.builder = function (yargs) {
     .example('$0 instrument ./lib ./output', 'instrument all .js files in ./lib with coverage and output in ./output')
 }
 
+const errorExit = msg => {
+  console.error(`nyc instrument failed:  ${msg}`)
+  process.exitCode = 1
+}
+
 exports.handler = function (argv) {
-  if (argv.output && (path.resolve(argv.cwd, argv.input) === path.resolve(argv.cwd, argv.output))) {
-    console.error(`nyc instrument failed: cannot instrument files in place, <input> must differ from <output>`)
-    process.exitCode = 1
-    return
+  if (argv.output && !argv.inPlace && (path.resolve(argv.cwd, argv.input) === path.resolve(argv.cwd, argv.output))) {
+    return errorExit(`cannot instrument files in place, <input> must differ from <output>.  Set '--in-place' to force`)
   }
 
   if (path.relative(argv.cwd, path.resolve(argv.cwd, argv.input)).startsWith('..')) {
-    console.error(`nyc instrument failed: cannot instrument files outside of project root directory`)
-    process.exitCode = 1
-    return
+    return errorExit('cannot instrument files outside project root directory')
+  }
+
+  if (argv.delete && argv.inPlace) {
+    return errorExit(`cannot use '--delete' when instrumenting files in place`)
   }
 
   if (argv.delete && argv.output && argv.output.length !== 0) {
     const relPath = path.relative(process.cwd(), path.resolve(argv.output))
     if (relPath !== '' && !relPath.startsWith('..')) {
       rimraf.sync(argv.output)
     } else {
-      console.error(`nyc instrument failed: attempt to delete '${process.cwd()}' or containing directory.`)
-      process.exit(1)
+      return errorExit(`attempt to delete '${process.cwd()}' or containing directory.`)
     }
   }
 
+  console.log(`Running nyc from ${process.cwd()}`)
```

Comment:
This will break streaming instrumentation, must be removed.  Example:
```sh
nyc instrument something.js > instrumented/something.js
```

Reply:
That's a bit embarrassing, that's some code I drop in while testing nyc from another project to confirm that my npm link has worked.  I think I should spend a little more time reviewing my code and not commit in such a hurry.

Thanks for the heads up Corey.

Reply:
Updating snapshots...

Reply:
It happens, this is why we do reviews :smile: